tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/arena/PureArena.scala

```diff
@@ -0,0 +1,248 @@
+package at.forsyte.apalache.tla.bmcmt.arena
+
+import at.forsyte.apalache.tla.bmcmt.types.{CellT, UnknownT}
+import at.forsyte.apalache.tla.bmcmt.{ArenaCell, CheckerException}
+import at.forsyte.apalache.tla.lir.{NameEx, TlaEx}
+
+/**
+ * An SMT-context free implementation of arenas.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+case class PureArena(
+    cellMap: Map[String, ArenaCell],
+    private val hasEdges: Map[ArenaCell, Seq[ElemPtr]],
+    private val domEdges: Map[ArenaCell, ArenaCell],
+    private val cdmEdges: Map[ArenaCell, ArenaCell],
```

Comment:
do we really need `cdmEdges` in new arenas?

Reply:
For now yes, because this implementation is interface-preserving.